firewood/src/db.rs

```diff
@@ -436,13 +436,23 @@ pub struct Db {
 impl Db {
     const PARAM_SIZE: u64 = size_of::<DbParams>() as u64;
 
-    /// Open a database.
-    pub fn new<P: AsRef<Path>>(db_path: P, cfg: &DbConfig) -> Result<Self, DbError> {
-        // TODO: make sure all fds are released at the end
+    pub async fn new<P: AsRef<Path> + Send + Sync + 'static>(
```

Comment:
This is over-constrained
```suggestion
    pub async fn new<P: AsRef<Path>>(
```